mux.go

```diff
@@ -358,6 +360,24 @@ func setCurrentRoute(r *http.Request, val interface{}) *http.Request {
 // Helpers
 // ----------------------------------------------------------------------------
 
+// getPath returns the escaped path if possible; doing what URL.EscapedPath()
+// which was added in go1.5 does
+func getPath(req *http.Request) string {
+	if req.RequestURI != "" {
+		// Extract the path from RequestURI (which is escaped unlike URL.Path)
+		// as detailed here as detailed in https://golang.org/pkg/net/url/#URL
+		// for < 1.5 server side workaround
+		// http://localhost/path/here?v=1 -> /path/here
+		iStart := len(req.URL.Scheme + `://` + req.URL.Host)
+		path := req.RequestURI[iStart:]
+		if i := strings.LastIndex(path, "?"); i > -1 {
```

Comment:
This handles URLs in the form `scheme://host/path?query` and `scheme://host/path?query#fragment`, but not `scheme://host/path#fragment`.